### PR TITLE
feat(tile): add support for adding highlight border via the  prop

### PR DESCRIPTION
**Pull request by edleeks87**
### Proposed behaviour

Adds support for rendering a highlight on the left border of the Tile component via the `highlightVariant` prop.
image

Fixes issue with separator having border radius
image

### Current behaviour

No support for highlight border
Last child has border radius
image

### Checklist

- [x] Commits follow our style guide
- [x] Screenshots are included in the PR if useful
- [x] Unit tests added or updated if required
- [x] Storybook added or updated if required
- [x] Typescript `d.ts` file added or updated if required

#### QA

- [ ] Tested in provided StackBlitz sandbox/Storybook
- [ ] Add new Playwright test coverage if required
- [ ] Carbon implementation matches Design System/designs
- [ ] UI Tests GitHub check reviewed if required

### Additional context

Using a pseudo selector causes `incomplete` axe warnings as it isn't able to discern the background colours and therefore cannot compute the contrast ratio

image

### CODE REVIEW

File: src/components/tile/tile.config.ts [hunk 1]
```diff
@@ -8,3 +8,11 @@ export const TILE_BORDER_VARIANTS = [
   "positive",
   "caution",
 ];
+export const TILE_HIGHLIGHT_VARIANTS = [
```
Comment (edleeks87) [inline src/components/tile/tile.config.ts:11]: I went back and forth about whether we should match the above border variant array elements, ticket steered to use Message variants but it feels a bit weird having the mismatch between the border and highlight variants etc (they do use slightly different tokens)

File: src/components/tile/tile.spec.tsx [hunk 2]
```diff
@@ -185,3 +187,16 @@ describe("Tile", () => {
     rootTagTest(wrapper.find(StyledTile), "tile", "foo", "bar");
   });
 });
+
+test.each<TileProps["highlightVariant"]>(TILE_HIGHLIGHT_VARIANTS)(
```
Comment (DipperTheDan) [inline src/components/tile/tile.spec.tsx:191]: **praise:** At least this is one test the person who picks up the Tile RTL refactor won't have to worry about.

**Review by nineteen88 — DISMISSED** [review]

**Review by DipperTheDan — COMMENTED** [review]
**issue:** It looks like the `with Accordion` example is missing the corners now.

Screenshot 2024-08-22 at 10 43 09

**Review by DipperTheDan — APPROVED** [review]

**Review by nineteen88 — APPROVED** [review]

File: src/components/tile/tile.stories.tsx [hunk 3]
```diff
@@ -1021,3 +1021,117 @@ export const Align: Story = () => {
...
+      <br />
+      <Tile highlightVariant="gradient" roundness="large">
+        <TileContent>Test Body One</TileContent>
+        <TileContent>Test Body Two</TileContent>
+        <TileContent>Test Body Three</TileContent>
+      </Tile>
+    </>
+  );
+};
+HighlightVariantGradientStory.storyName = "Highlight variant gradient story";
```
Comment (nineteen88) [inline src/components/tile/tile.stories.tsx:1042]: nitpick(non-blocking): inconsistent casing on new story names

**Review by paulrobinson88 — APPROVED** [review]

**Pull request merged**